### PR TITLE
Show release version on both apps and new release 

### PR DESCRIPTION
**Pull request by santhoshb-msft**
1. Create Classes to store AppVersion and latest version from Git
2. Logic to fetch the new release from this repo
3. Show the release number on both customer app, admin app and metering project
4. Compare the current release with new release and add msg to Admin-home on new release available message.

### CODE REVIEW

File: src/Services/Services/AppVersionService.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
+﻿using System.Diagnostics;
+using System.Net.Http;
+using System.Text.Json;
+using System.Threading.Tasks;
+
+namespace Marketplace.SaaS.Accelerator.Services.Services;
+
+/// <summary>
+/// AppVersionService Service.
```
Comment (007zszmz) [inline src/Services/Services/AppVersionService.cs:9]: provide an example of the version in the comment.

File: src/AdminSite/Controllers/HomeController.cs [hunk 2]
```diff
@@ -210,6 +236,15 @@ public IActionResult Index()
         {
             this.applicationConfigService.SaveFileToDisk("LogoFile", "contoso-sales.png");
             this.applicationConfigService.SaveFileToDisk("FaviconFile", "favicon.ico");
+            var newReleaseVersion = this.sAGitReleasesService.GetLatestRelease().GetAwaiter().GetResult();
```
Comment (007zszmz) [inline src/AdminSite/Controllers/HomeController.cs:239]: why not make GetLatestRelease synchronized method so you don't need to block the thread?

File: src/Services/Services/SAGitReleasesService.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+/// <summary>
+/// SAGitReleasesService Service.
+/// </summary>
+public class SAGitReleasesService : ISAGitReleasesService
+{
+    /// <summary>
+    /// Gets the latest release number.
+    /// </summary>
+    /// <returns> Release Version.</returns>
+    public async Task<string> GetLatestRelease()
```
Comment (007zszmz) [inline src/Services/Services/SAGitReleasesService.cs:16]: by using HttpClient.Send and HttpContent.ReadAsStream, you can make this method synchronous. And you don't need to call GetResult(), if that's the only place this is used.

File: src/Services/Services/SAGitReleasesService.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    {
+        using (var httpClient = new HttpClient())
+        {
+            // GitHub API requires a user agent
+            httpClient.DefaultRequestHeaders.UserAgent.TryParseAdd("request");
+
```
Comment (007zszmz) [inline src/Services/Services/SAGitReleasesService.cs:22]: handle http exceptions.

File: src/Services/Services/AppVersionService.cs (hunk 1 above)
Comment (santhoshb-msft) [reply src/Services/Services/AppVersionService.cs:9]: image
image
image

File: src/Services/Services/SAGitReleasesService.cs (hunk 3 above)
Comment (santhoshb-msft) [reply src/Services/Services/SAGitReleasesService.cs]: @007zszmz this should be fixed now.

File: src/Services/Services/SAGitReleasesService.cs (hunk 4 above)
Comment (santhoshb-msft) [reply src/Services/Services/SAGitReleasesService.cs:36]: @007zszmz this should be fixed now.

File: src/AdminSite/Controllers/HomeController.cs (hunk 2 above)
Comment (santhoshb-msft) [reply src/AdminSite/Controllers/HomeController.cs]: @007zszmz this should be fixed now.

**Comment by santhoshb-msft** [comment]
@007zszmz can you please review on your comments and help with an approval.

**Comment by santhoshb-msft** [comment]
@neelavarshad @007zszmz can you help approve this please?

**Comment by neelavarshad** [comment]
@santhoshb-msft  Is this expected for first publish? 
image

**Comment by santhoshb-msft** [comment]
> @santhoshb-msft Is this expected for first publish? image

Yes @neelavarshad this is expected, if someone deploys any branch other than the release, we show this. We could probably make it 0.0.0 if need be.

**Review by neelavarshad — APPROVED** [review]
Approved all changes

**Pull request merged**